src/main/java/org/folio/rest/impl/ItemStorageAPI.java

```diff
@@ -143,22 +144,15 @@ public void deleteItemStorageItemsByItemId(
 
   private CompletableFuture<Item> setEffectiveCallNumber(Map<String, String> okapiHeaders, Context vertxContext, Item item) {
     CompletableFuture<Item> completableFuture = null;
-    EffectiveCallNumberComponents components = new EffectiveCallNumberComponents();
-    if (item.getItemLevelCallNumber() != null && !item.getItemLevelCallNumber().isEmpty()) {
-      components.setCallNumber(item.getItemLevelCallNumber());
-      item.setEffectiveCallNumberComponents(components);
+    if (shouldNotRetreiveHoldingsRecord(item)) {
+      item.setEffectiveCallNumberComponents(EffectiveCallNumberComponentsUtil.buildComponents(null, item));
       completableFuture = CompletableFuture.supplyAsync(() -> item);
     } else {
-      if (item.getHoldingsRecordId() != null && !item.getHoldingsRecordId().isEmpty()) {
-        completableFuture = getHoldingsRecordById(okapiHeaders, vertxContext, item.getHoldingsRecordId()).thenCombineAsync(CompletableFuture.supplyAsync(() -> item), (hr, i) ->
-        {
-          components.setCallNumber(hr.getCallNumber());
-          i.setEffectiveCallNumberComponents(components);
-          return i;
-        });
-      } else {
-        completableFuture = CompletableFuture.supplyAsync(() -> item);
-      }
+      completableFuture = getHoldingsRecordById(okapiHeaders, vertxContext, item.getHoldingsRecordId()).thenCombineAsync(CompletableFuture.supplyAsync(() -> item), (hr, i) ->
+      {
+        i.setEffectiveCallNumberComponents(EffectiveCallNumberComponentsUtil.buildComponents(hr, item));
```

Comment:
Why is there a need to refer to `i` and `item` are these not the same item?

Reply:
It seems like it is available in the context of the lambda function so I'm not sure why it's being passed in like that. It seems to work without supplying it, so I think we can remove the redundancy altogether.